### PR TITLE
bismark: don't use iGenomes index with hisat2

### PR DESCRIPTION
**Pull request by phue**
Proposed fix for https://github.com/nf-core/methylseq/issues/207

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/methylseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by ewels — CHANGES_REQUESTED** [review]
Nice idea, but I think that `if( params.aligner =~ /bismark/ ){` matches both `bismark` and `bismark_hisat` so moving the code doesn't actually make much difference 😓

File: main.nf [hunk 1]
```diff
@@ -51,6 +48,7 @@ Channel
 ch_splicesites_for_bismark_hisat_align = params.known_splices ? Channel.fromPath(params.known_splices, checkIfExists: true) : Channel.empty()
 
 if( params.aligner =~ /bismark/ ){
+    params.bismark_index = params.genome && params.aligner == 'bismark' ? params.genomes[ params.genome ].bismark ?: false : false
```
Comment (phue) [inline main.nf:51]: The relevant change is here

Comment (phue) [reply main.nf:51]: So with this, the iGenomes index is only used if `params.aligner == bismark` (which is using `bowtie2`)

Comment (ewels) [reply main.nf:51]: Ahh sorry, I didn't notice that it had changed, my bad. Great!

**Review by ewels — APPROVED** [review]

**Pull request merged**